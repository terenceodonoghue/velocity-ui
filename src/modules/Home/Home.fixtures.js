export const customizedLabel = {
  index: 0,
  percent: 86,
};
