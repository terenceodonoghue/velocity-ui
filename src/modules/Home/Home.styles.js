import styled from '@emotion/styled';
import { Container, variables } from '../../components';

const { colors, typography } = variables;

export const Heading = styled.h2`
  margin: 0 0 13px;
  width: 135px;
  font-size: 28px;
  font-weight: ${typography.fontWeightLight};
  line-height: 1.14;
`;

export const Paragraph = styled.p`
  margin: 0;
  color: ${colors.blueyGrey};
  line-height: 1.47;
`;

export const Text = styled.div`
  padding: 45px 0 42px 61px;
`;

export const Welcome = styled(Container.Card)`
  display: flex;
  height: 230px;
  padding: 0 27px 0 46px;
  width: 558px;
`;
