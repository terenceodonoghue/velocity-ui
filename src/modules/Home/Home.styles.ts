import { css, InterpolationWithTheme } from '@emotion/core';
import { Theme } from '../../components';

export const welcome = css`
  display: flex;
  height: 230px;
  padding: 0 27px 0 46px;
  width: 558px;
`;

export const welcomeContent = css`
  padding: 45px 0 42px 61px;
`;

export const welcomeHeading: InterpolationWithTheme<Theme> = (theme) => css`
  margin: 0 0 13px;
  width: 135px;
  font-size: 28px;
  font-weight: ${theme.typography.fontWeightLight};
  line-height: 1.14;
`;

export const welcomeText: InterpolationWithTheme<Theme> = (theme) => css`
  margin: 0;
  color: ${theme.colors.blueyGrey};
  line-height: 1.47;
`;
