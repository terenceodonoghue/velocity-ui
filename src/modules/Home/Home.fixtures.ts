import faker from 'faker';
import { TopDriversProps } from './Home';

export const customizedLabel = {
  index: 0,
  percent: 86,
};

export const topDrivers: TopDriversProps = {
  drivers: [
    {
      src: faker.image.avatar(),
      name: 'Bebop',
      vehicle: 'Volvo Intellisafe',
      earnings: 6432,
      distance: 1232,
    },
    {
      src: faker.image.avatar(),
      name: 'Gran Tesoro',
      vehicle: 'Chevrolet Bolt',
      earnings: 5342,
      distance: 945,
    },
  ],
};
