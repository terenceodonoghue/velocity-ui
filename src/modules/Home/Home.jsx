import React from 'react';
import { number } from 'prop-types';
import {
  PieChart, Pie, ResponsiveContainer,
} from 'recharts';
import * as S from './Home.styles';

const outerTicks = [
  { name: 'score', value: 86, fill: '#2e5bff' },
  { name: 'max', value: 14, fill: '#e0e7ff' },
];

const innerTicks = [
  { name: 'score', value: 100, fill: '#e0e7ff' },
];

export const renderCustomizedLabel = ({
  index, percent,
}) => (
  index === 0 && (
    <>
      <text fill="#2e384d" fontFamily="Rubik-Light, Rubik" fontSize="48" fontWeight="300" letterSpacing="-0.600000024" x="50%" y="50%">
        <tspan x="75" y="140">{`${(percent * 100).toFixed(0)}`}</tspan>
      </text>
      <text
        dominantBaseline="central"
        fill="#8798ad"
        fontFamily="Rubik-Regular, Rubik"
        fontSize="13"
        fontWeight="normal"
        letterSpacing="1.213333"
        x="50%"
        y="50%"
      >
        <tspan x="62" y="162">OPERATING</tspan>
        <tspan x="78" y="177">SCORE</tspan>
      </text>
    </>
  )
);

const Welcome = () => (
  <S.Welcome>
    <ResponsiveContainer width={203.5}>
      <PieChart>
        <Pie cx="50%" cy="64%" data={innerTicks} dataKey="value" innerRadius={66} outerRadius={69} startAngle={190} endAngle={-10} />
        <Pie
          cx="50%"
          cy="64%"
          data={outerTicks}
          dataKey="value"
          paddingAngle={3}
          innerRadius={75}
          outerRadius={101}
          startAngle={190}
          endAngle={-10}
          label={renderCustomizedLabel}
          labelLine={false}
        />
      </PieChart>
    </ResponsiveContainer>
    <S.Text>
      <S.Heading>Welcome to Velocity</S.Heading>
      <S.Paragraph>
        All cars are operating well. There were 1,233 trips since your last login.
      </S.Paragraph>
    </S.Text>
  </S.Welcome>
);

renderCustomizedLabel.propTypes = {
  index: number.isRequired,
  percent: number.isRequired,
};

export default {
  Welcome,
};
