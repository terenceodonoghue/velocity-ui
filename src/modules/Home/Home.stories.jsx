import React from 'react';
import { storiesOf } from '@storybook/react';
import { Welcome } from './Home';

storiesOf('Modules|Home', module).add('Welcome', () => <Welcome />);
