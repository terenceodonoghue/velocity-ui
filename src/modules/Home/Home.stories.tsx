import React, { FunctionComponent } from 'react';
import * as Home from './Home';
import * as fixtures from './Home.fixtures';

export default {
  title: 'Modules/Home',
};

export const TopDrivers: FunctionComponent = () => (
  <Home.TopDrivers {...fixtures.topDrivers} />
);

export const Welcome: FunctionComponent = () => <Home.Welcome />;
