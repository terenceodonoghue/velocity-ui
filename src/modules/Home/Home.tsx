/** @jsx jsx */
import { jsx } from '@emotion/core';
import { Fragment, FunctionComponent } from 'react';
import {
  ContentRenderer,
  Pie,
  PieChart,
  PieLabelRenderProps,
  ResponsiveContainer,
} from 'recharts';
import { Avatars, Surfaces } from '~/components';
import * as css from './Home.styles';

interface Driver {
  src: string;
  name: string;
  vehicle: string;
  earnings: number;
  distance: number;
}

export interface TopDriversProps {
  drivers: Driver[];
}

const outerTicks = [
  { name: 'score', value: 86, fill: '#2e5bff' },
  { name: 'max', value: 14, fill: '#e0e7ff' },
];

const innerTicks = [{ name: 'score', value: 100, fill: '#e0e7ff' }];

export const renderCustomizedLabel:
  | ContentRenderer<PieLabelRenderProps>
  | boolean = ({ index, percent = 0 }) =>
  index === 0 && (
    <Fragment>
      <text
        fill="#2e384d"
        fontFamily="Rubik-Light, Rubik"
        fontSize="48"
        fontWeight="300"
        letterSpacing="-0.600000024"
        x="50%"
        y="50%"
      >
        <tspan x="75" y="140">{`${(percent * 100).toFixed(0)}`}</tspan>
      </text>
      <text
        dominantBaseline="central"
        fill="#8798ad"
        fontFamily="Rubik-Regular, Rubik"
        fontSize="13"
        fontWeight="normal"
        letterSpacing="1.213333"
        x="50%"
        y="50%"
      >
        <tspan x="62" y="162">
          OPERATING
        </tspan>
        <tspan x="78" y="177">
          SCORE
        </tspan>
      </text>
    </Fragment>
  );

export const TopDrivers: FunctionComponent<TopDriversProps> = ({ drivers }) => (
  <Surfaces.Card heading="Top Drivers">
    {drivers.map(({ distance, earnings, name, src, vehicle }, i) => (
      <div>
        <div css={css.topDriverImg(i + 1)}>
          <Avatars.Round css={css.topDriversAvatar} src={src} />
        </div>
        <span>{name}</span>
        <span>{earnings}</span>
        <span>{vehicle}</span>
        <span>{distance}</span>
      </div>
    ))}
  </Surfaces.Card>
);

export const Welcome: FunctionComponent = () => (
  <Surfaces.Card css={css.welcome}>
    <ResponsiveContainer width={203.5}>
      <PieChart>
        <Pie
          cx="50%"
          cy="64%"
          data={innerTicks}
          dataKey="value"
          innerRadius={66}
          outerRadius={69}
          startAngle={190}
          endAngle={-10}
        />
        <Pie
          cx="50%"
          cy="64%"
          data={outerTicks}
          dataKey="value"
          paddingAngle={3}
          innerRadius={75}
          outerRadius={101}
          startAngle={190}
          endAngle={-10}
          label={renderCustomizedLabel}
          labelLine={false}
        />
      </PieChart>
    </ResponsiveContainer>
    <div css={css.welcomeContent}>
      <h2 css={css.welcomeHeading}>Welcome to Velocity</h2>
      <p css={css.welcomeText}>
        All cars are operating well. There were 1,233 trips since your last
        login.
      </p>
    </div>
  </Surfaces.Card>
);
