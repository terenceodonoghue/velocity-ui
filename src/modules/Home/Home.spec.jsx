import React from 'react';
import Home from './Home';
import { customizedLabel } from './Home.fixtures';

describe('Home', () => {
  describe('Welcome', () => {
    it('matches snapshot', () => {
      const tree = renderer.create(
        <Home.Welcome />,
        {
          createNodeMock: () => (
            { parentElement: document.implementation.createHTMLDocument().body }
          ),
        },
      ).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });

  describe('renderCustomizedLabel', () => {
    it('matches snapshot', () => {
      const tree = renderer.create(
        <Home.renderCustomizedLabel {...customizedLabel} />,
      ).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
});
