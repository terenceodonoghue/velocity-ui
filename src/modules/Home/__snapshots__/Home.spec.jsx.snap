// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Welcome matches snapshot 1`] = `
.emotion-7 {
  display: block;
  border-radius: 1px;
  border: solid 1px rgba(46,91,255,0.08);
  box-shadow: 0 10px 20px 0 rgba(46,91,255,0.07);
  padding: 32px 24px;
  background-color: var(--color-white);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 230px;
  padding: 0 27px 0 46px;
  width: 558px;
}

.emotion-4 {
  padding: 45px 0 42px 61px;
}

.emotion-0 {
  margin: 0 0 13px;
  width: 135px;
  font-size: 28px;
  font-weight: var(--font-weight-light);
  line-height: 1.14;
}

.emotion-2 {
  margin: 0;
  color: var(--color-bluey-grey);
  line-height: 1.47;
}

<div
  className="emotion-6 emotion-7 emotion-8"
>
  <div
    className="recharts-responsive-container"
    style={
      Object {
        "height": "100%",
        "maxHeight": undefined,
        "minHeight": undefined,
        "minWidth": undefined,
        "width": 203.5,
      }
    }
  >
    <div
      style={
        Object {
          "display": "none",
          "height": 0,
          "position": "absolute",
          "visibility": "hidden",
          "width": 0,
        }
      }
    />
  </div>
  <div
    className="emotion-4 emotion-5"
  >
    <h2
      className="emotion-0 emotion-1"
    >
      Welcome to Velocity
    </h2>
    <p
      className="emotion-2 emotion-3"
    >
      All cars are operating well. There were 1,233 trips since your last login.
    </p>
  </div>
</div>
`;

exports[`Home renderCustomizedLabel matches snapshot 1`] = `
Array [
  <text
    fill="#2e384d"
    fontFamily="Rubik-Light, Rubik"
    fontSize="48"
    fontWeight="300"
    letterSpacing="-0.600000024"
    x="50%"
    y="50%"
  >
    <tspan
      x="75"
      y="140"
    >
      8600
    </tspan>
  </text>,
  <text
    dominantBaseline="central"
    fill="#8798ad"
    fontFamily="Rubik-Regular, Rubik"
    fontSize="13"
    fontWeight="normal"
    letterSpacing="1.213333"
    x="50%"
    y="50%"
  >
    <tspan
      x="62"
      y="162"
    >
      OPERATING
    </tspan>
    <tspan
      x="78"
      y="177"
    >
      SCORE
    </tspan>
  </text>,
]
`;
